color/src/temperature.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025 the Color Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#![expect(
+    clippy::excessive_precision,
+    reason = "We are sticking with the constants from Tanner Helland."
+)]
+
+use crate::{AlphaColor, ColorSpace, OpaqueColor};
+
+#[cfg(all(not(feature = "std"), not(test)))]
+use crate::floatfuncs::FloatFuncs;
+
+// Implementation notes:
+//
+// * These can't be const yet as `clamp`, `ln`, and `powf` are not `const`.
+
+impl<CS: ColorSpace> AlphaColor<CS> {
+    /// Convert a temperature in Kelvin to a color.
+    ///
+    /// The Kelvin temperature will be clamped to a range of 1000-40,000.
```

Comment:
The docs should be improved here and would be better with some color swatches and a table.

Also, need to credit the original blog post and link to Wikipedia about color temperature.

Reply:
Hmm, it is suspect to me that the calculations clamp to the sRGB gamut, but go to 1000 kelvin. I believe temperatures lower than roughly 2000 kelvin are outside sRGB's natural gamut, and for e.g. 1667 kelvin, it should be roughly  `color(srgb 1.0 0.46 -0.12)`, based on the chromaticities here https://en.wikipedia.org/w/index.php?title=Planckian_locus&oldid=1245078623#Approximation.